server/src/main/java/org/apache/iotdb/db/service/TSServiceImpl.java

```diff
@@ -523,7 +529,17 @@ public TSExecuteStatementResp executeStatement(TSExecuteStatementReq req) {
       PhysicalPlan physicalPlan;
       physicalPlan = processor.parseSQLToPhysicalPlan(statement, zoneIds.get());
       if (physicalPlan.isQuery()) {
-        return executeQueryStatement(statement, physicalPlan);
+      	resp = executeQueryStatement(statement, physicalPlan);
+      	long endtime = System.currentTimeMillis();
+      	sqlArgument = new SqlArgument(resp,physicalPlan,statement,starttime,endtime);
+      	sqlArgument.setStatement(statement);
+      	sqlArgumentsList.add(sqlArgument);
+      	if(sqlArgumentsList.size()>200) {
+  			for (int i = 0; i < 50; i++) {
+  				sqlArgumentsList.remove(sqlArgumentsList.size()-1);
```

Comment:
It seems that these lines are used to limit total size of saved sqls. But this line seems removes the latest 50 sqls rather than the oldest ones.